src/firstboot/firstboot.c

```diff
@@ -667,6 +721,31 @@ static int process_root_password(void) {
         if (lock < 0)
                 return log_error_errno(lock, "Failed to take a lock: %m");
 
+        if (arg_delete_root_password) {
+                const char *etc_passwd;
+                struct passwd *p;
+
+                if (arg_copy_root_password || arg_root_password || arg_prompt_root_password)
+                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                               "--delete-root-password cannot be combined with other root password options");
+
+                etc_passwd = prefix_roota(arg_root, "/etc/passwd");
+
+                p = getpwnam("root");
```

Comment:
hmmm, we shouldn't use getpwnam(), since that operates on the host /etc/passwd (i.e. doesn't take arg_root into account). I figure we need to use fgetpwent_sane() for reading this... Maybe instead just change the write_root_passwd() function to take the password string instead of the full struct passwd, and then have it patch that in as it goes.